ale_linters/rust/cargo.vim

```diff
@@ -3,19 +3,89 @@
 
 
 function! ale_linters#rust#cargo#GetCargoExecutable(bufnr) abort
-    if ale#util#FindNearestFile(a:bufnr, 'Cargo.toml') !=# ''
-        return 'cargo'
-    else
-        " if there is no Cargo.toml file, we don't use cargo even if it exists,
-        " so we return '', because executable('') apparently always fails
-        return ''
-    endif
+	if ale#util#FindNearestFile(a:bufnr, 'Cargo.toml') !=# ''
+		return 'cargo'
+	else
+		" if there is no Cargo.toml file, we don't use cargo even if it exists,
+		" so we return '', because executable('') apparently always fails
+		return ''
+	endif
+endfunction
+
+function! ale_linters#rust#cargo#HandleCargoErrors(buffer_number, errorlines) abort
```

Comment:
The indentation here makes this hard to read. How does this differ from the rustc handler?